hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeStateMachine.java

```diff
@@ -704,6 +711,44 @@ private Thread getCommandHandlerThread(Runnable processCommandQueue) {
     return handlerThread;
   }
 
+  /**
+   * Stops the DiskBalancerService if it is running.
+   */
+  public void stopDiskBalancer() {
+    OzoneContainer ozoneContainer = getContainer();
+    if (ozoneContainer != null) {
+      DiskBalancerService diskBalancerService = ozoneContainer.getDiskBalancerService();
+      if (diskBalancerService != null) {
+        shouldRun = diskBalancerService.getDiskBalancerInfo().isShouldRun();
+        paused.set(true); // Mark as paused
+        diskBalancerService.setShouldRun(false);
+      }
+    }
+  }
+
+  /**
+   * Resume the DiskBalancerService if it was running previously.
+   */
+  public void resumeDiskBalancer() {
+    OzoneContainer ozoneContainer = getContainer();
+    if (ozoneContainer != null) {
+      DiskBalancerService diskBalancerService = ozoneContainer.getDiskBalancerService();
+      if (diskBalancerService != null && paused.get()) {
+        // Reset paused state
+        paused.set(false);
+        diskBalancerService.setShouldRun(true);
+      }
+    }
+  }
+
+  /**
+   * Checks if DiskBalancerService should run based
+   * on its previous state and paused flag.
+   */
+  public boolean shouldRunDiskBalancer() {
+    return shouldRun && paused.get();
```

Comment:
Based on the above, shouldRun can be removed here. Just use paused.get() directly.